docs: add onboarding guide for identifying good first issues (#1815)
asyncapi/community: docs/onboarding-guide/identifying-good-first-issues.md

To find good first issues, follow these steps:

1. Navigate to the [AsyncAPI GitHub organization](https://github.com/asyncapi).

![Screenshot 2025-05-24 020543](https://github.com/user-attachments/assets/13573e2d-8d30-460f-8923-fcad82eeaedc)

2. Click on the **"Issues"** tab of a repository.

![Screenshot 2025-05-24 020327](https://github.com/user-attachments/assets/8f343ae0-2206-4be9-b1e6-9cd6ec192ef8)

3. Use the search bar or label filters to locate issues labeled like: `good first issue` or `help wanted`.

![Screenshot 2025-05-24 020450](https://github.com/user-attachments/assets/49dab98c-2b51-4374-8abe-130c2c269110)

These labels indicate tasks that are ideal starting points for new contributors. Think of them as Level 1 quests - easy enough to get started, but important to the project's progress.

### Understanding Issue Labels
When looking at issues, you might see different labels. Here's what they mean:

| Label            | Meaning                                      |
|---------------------|-----------------------------------------------|
| **good first issue** | Great for newcomers! Clearly defined and simple. |

> | **help wanted**      | Extra attention is needed, may require some research. |

Reviewer: Lines 22-25, 43, 50-51 repeat the same information over and over again

> # Start Contributing to AsyncAPI

Reviewer 2: Please add front matter

> Welcome to the AsyncAPI community! Not sure where to make your first contribution? This guide is a treasure map to help you navigate the process, find the right issues, and get started with contributing to AsyncAPI.

Author: The front matter requires the description too

---
title: Identifying Good First Issues
description: A guide to help new contributors find and work on beginner-friendly issues in AsyncAPI projects
weight: 50
---

> # Start Contributing to AsyncAPI

Reviewer: @SaxenaAnushka102 Please fix this as well 👆

title: Identifying Good First Issues
sidebar_label: Identifying Good First Issues
description: A guide to help new contributors find and work on beginner-friendly issues in AsyncAPI projects
weight: 50
---

# Start Contributing to AsyncAPI
Welcome to the AsyncAPI community! Not sure where to make your first contribution? This guide is a treasure map to help you navigate the process, find the right issues, and get started with contributing to AsyncAPI.

## Your First Contribution
### Find Something to Work On
The first step in contributing to AsyncAPI is identifying an issue to work on. Contributions come in many forms, and every contribution, big or small, matters! Here are a few ways to get started:

- Improve the documentation (fix typos, enhance explanations, add examples).

- Clarify code comments, improve variable names, or refactor code.

- Add test coverage for untested areas.

- Help triage issues (identify duplicates, provide additional context).

If none of the above resonate with you, check out issues labeled `"good first issue"` or `"help wanted"` in the repositories. These labels indicate tasks that are well-scoped for new contributors.

> >⚠ Note: While all contributions are welcome, each pull request requires review and testing from maintainers. Please avoid unnecessary PRs (like fixing minor linter warnings, single-character typos or reformatting) unless they add real value. Always communicate with maintainers before making such changes.

Author: Done👍

## How to Find Good First Issues
Imagine you're a detective searching for clues. In open-source, your clues are GitHub issues— tasks that need solving. Some issues are complex mysteries, while others are simple puzzles that are perfect for beginners.

### Look for "Good First Issue" Labels
GitHub repositories often label issues to indicate difficulty levels. For beginners, the best ones to start with are:

- good first issue – These are beginner-friendly and usually come with clear instructions.
- help wanted – These issues need attention, but may require a bit more effort.

### How to Find These Issues:
1.  Go to the [AsyncAPI repository](https://github.com/asyncapi) you’re interested in, like [`AsyncAPI Generator`](https://github.com/asyncapi/generator).
2. Click on the "Issues" tab .
3. Use the search bar to filter beginner-friendly issues:

```bash
label:"good first issue" is:open
```
or

```bash
label:"help wanted" is:open
```
4. Read through the issue descriptions and pick one that interests you.

> This is like a video game where "Good First Issues" are Level 1 quests— easy enough for beginners but still important for the game progression!

Reviewer: ```suggestion
This is like a video game where "Good First Issues" are Level 1 quests - easy enough for beginners but still important for the game progression.
```

### Understanding Issue Labels

> When looking at issues, you might see different labels. Here’s what they mean:

Reviewer: ```suggestion
When looking at issues, you might see different labels. Here's what they mean:
```

| Label            | Meaning                                      |
|---------------------|-----------------------------------------------|
| **good first issue** | Great for newcomers! Clearly defined and simple. |
| **help wanted**      | Extra attention is needed, may require some research. |

> | **bug**             | Something isn't working correctly—time to fix it!|

Reviewer: ```suggestion
| **bug**             | Something isn't working correctly, so it's time to fix it!|
```

| **enhancement**     | Suggests a new feature or improvement. |
| **📑 docs**         | Documentation-related issues. |
| **🐞 docs bug**     | Documentation errors or inconsistencies. |
| **🎨 design**       | Design thinking/mockups needed. |
| **area/docs**       | Issues related to documentation. |
| **area/design**     | Issues related to design improvements. |
| **area/ci-cd**      | Issues related to CI/CD pipelines. |
| **area/go**         | Issues specific to Go projects. |
| **area/java**       | Issues specific to Java projects. |
| **area/javascript** | Issues specific to JavaScript projects. |
| **area/typescript** | Issues specific to TypeScript projects. |
| **bounty**          | AsyncAPI Bounty program-related issue. |
| **gsoc**           | Google Summer of Code-related ideas/issues. |
| **gsod**           | Google Season of Docs-related ideas/issues. |
| **Hacktoberfest**   | Available for Hacktoberfest participants. |
| **dependencies**    | Pull requests that update dependencies. |

> | **do-not-merge**    | Indicates PRs that shouldn't be merged.|

Reviewer: What are possible use cases here?

Author: GSoD is not happening anymore so I've removed that. Remaining ones I've collected from repos where hacktoberfest issues are created. My idea was to give a brief description of how can a beginner know which issue label means what as labels might differ from org to org.

| **Epic**           | Large issues that track multiple related tasks. |
| **level/advanced**  | Issues requiring advanced knowledge. |
| **level/medium**    | Issues requiring intermediate skills. |
| **invalid**        | This issue or PR doesn't seem valid.|
| **duplicate**      | This issue or PR already exists.|
| **keep-open**      | Issue should remain open for tracking.  |

### Next Steps

> AsyncAPI follows a "first come, first served" approach for issue assignment:
> 
> - Comment on the issue – Start with saying something like, "Hey, I’d like to work on this!" so maintainers know you're interested.
> - We do not pre-assign issues. Instead, contributors must comment on an issue they want to work on.
> - Once you comment, wait for approval by maintainers. They may guide you further (if needed) or might assign the issue to you. If there are no blockers, you can start working!
> - Start coding! – Once approved, fork the repository, create a new branch, and start working!
> - Ask for help – Don’t hesitate to ask in the issue comments or the dedicated AsyncAPI Slack Channel if you’re stuck.

Reviewer 3: You're off to a great start! :) Just gave some ideas on how to make this portion a bit more readable 
```suggestion
AsyncAPI follows a "first come, first served" approach for assigning issues. Here are some tips to help you increase your chances of getting assigned to an issue:

- **Comment on the issue:** Start with saying something like, "Hey, I’d like to work on this!" so maintainers know you're interested.
>⚠ Note: We do not pre-assign issues. Instead, contributors must comment on an issue they want to work on.
- **Once you comment, wait for approval by maintainers:** They may guide you further (if needed) or might assign the issue to you. If there are no blockers, you can start working!
- **Start coding**: Once approved, [fork the repository](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo), [create a new branch](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging), and start working!
- **Ask for help**: Don’t hesitate to ask in the issue comments or the dedicated AsyncAPI Slack Channel if you’re stuck.
```

Reviewer: @CBID2 On line 92, I would suggest linking to the [Git Workflow](https://github.com/asyncapi/community/blob/master/git-workflow.md) document as it contains instructions specific to AsyncAPI instead of generic ones.

Author: Thanks for the reviews and the detailed explanation to all the changes, @CBID2 and @bandantonio 🙏

> So what are you waiting for? Go find your first quest, and let’s build something awesome together! 

Reviewer: ```suggestion
So what are you waiting for? Go find your first quest, and let's build something awesome together! 
```
